index.bs

```diff
@@ -5483,6 +5500,415 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+    : [=public key credential source/type=]
+    :: {{public-key}}
+    : [=public key credential source/id=]
+    :: |credentialId|
+    : [=public key credential source/privateKey=]
+    :: |privateKey|
+    : [=public key credential source/rpId=]
+    :: |rpId|
+    : [=public key credential source/userHandle=]
+    :: |userHandle|
+ 1. Associate a [=signature counter=] |counter| to the |credential| with a starting value equal to the |parameters|'
+     |signCount| or `0` if |signCount| is `null`.
+ 1. Store the |credential| and |counter| in the database of the |authenticator|.
+ 1. Return [=success=].
+
+## <dfn>Get Credentials</dfn> ## {#sctn-automation-get-credentials}
+
+The [=Get Credentials=] WebDriver [=extension command=] returns one [=Credential Parameters=] object for every
+[=Public Key Credential Source=] stored in a [=Virtual Authenticator=], regardless of whether they were
+stored using [=Add Credential=] or {{CredentialsContainer/create()|navigator.credentials.create()}}. It is defined as follows:
+
+<figure id="table-getCredentials" class="table">
+    <table class="data">
+        <thead>
+            <tr>
+                <th>HTTP Method</th>
+                <th>URI Template</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td>GET</td>
+                <td>`/session/{session id}/webauthn/authenticator/{authenticatorId}/credentials`</td>
+            </tr>
+        </tbody>
+    </table>
+</figure>
+
+The [=remote end steps=] are:
+
+ 1. If |authenticatorId| does not match any [=Virtual Authenticator=] stored in the [=Virtual Authenticator
+     Database=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Let |credentialsArray| be an empty array.
+ 1. For each [=Public Key Credential Source=] |credential|, managed by the authenticator identified by |authenticatorId|,
+    construct a corresponding [=Credential Parameters=] [=Object=] and add it to |credentialsArray|.
+ 1. Return [=success=] with data containing |credentialsArray|.
+
+## <dfn>Remove Credential</dfn> ## {#sctn-automation-remove-credential}
+
+The [=Remove Credential=] WebDriver [=extension command=] removes a [=Public Key Credential Source=] stored on a
+[=Virtual Authenticator=]. It is defined as follows:
+
+<figure id="table-removeCredential" class="table">
+    <table class="data">
+        <thead>
+            <tr>
+                <th>HTTP Method</th>
+                <th>URI Template</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td>DELETE</td>
+                <td>`/session/{session id}/webauthn/authenticator/{authenticatorId}/credentials/{credentialId}`</td>
+            </tr>
+        </tbody>
+    </table>
+</figure>
+
+The [=remote end steps=] are:
+
+ 1. If |authenticatorId| does not match any [=Virtual Authenticator=] stored in the [=Virtual Authenticator
```

Comment:
Do we really need to error here when there are not items that match. Can't we just try do the right thing and if not just go meh?

Reply:
What do you mean exactly? If the `authenticatorId` doesn't match, just don't do anything and return `success`? If that's the case, that would potentially be a source of confusion for developers that's easily avoided by returning an error... the UA has to check for the existance of the authenticator anyway. Maybe I'm misunderstanding you?

Reply:
Since we are deleting it, if it is not there does it really matter is what I mean. If you feel it does matter then leave it in.

Reply:
I think it does matter. The most likely scenario for trying to delete something that is not there is a developer oversight. Catching that early means less headaches. And if say a library wanted that functionality, they could catch the error and pretend nothing happened.

Reply:
This and the above conversation can be set as resolved now?